### ISSUE TITLE
Mesh Agent is getting crashed (MeshCentral version 1.1.0)

### ISSUE THREAD

**Issue by sukeshleel** [Ylianst/MeshAgent#174]
Mesh agent is getting crashed every 10mins once, there is no anti virus application available and windows defender is also disabled, 

Tried to run the mesh agent in command prompt to find the cause but after 10mins its just getting exited not able to find any info from command prompt why crash happening. So, enabled the debug log but couldn't find error for the root cause of this issue.

Attaching the screenshots for the reference. Kindly let me know what causing the crash 
![EventViewerError](https://user-images.githubusercontent.com/33314744/220073225-fc13b2ae-b689-417f-9939-29745abf0ade.png)
[Debuglog.txt](https://github.com/Ylianst/MeshAgent/files/10783148/Debuglog.txt)

**Comment by knkumar93** [comment]
Hi @sukeshleel , Similar issue reported in #130 . Try once by enabling core dump in meshagent.msh file, and see if crash dump generates when crash happens. In my case, crash dump was not generating.
`coreDumpEnabled=1`

**Comment by sukeshleel** [comment]
Hi @knkumar93, I have tried enabling the coreDumpEnabled=1, mesh agent is crashing but dump file is not generating.

Hi @Ylianst and @krayon007 , kindly let me know what can be done here to check on this issue.

**Comment by PrplHaz4** [comment]
I'm seeing these fairly regularly too.  I'm working to correlate to other events but nothing is jumping out at me yet.

![image](https://github.com/Ylianst/MeshAgent/assets/841414/811934ac-7792-4335-a7cc-9d0b861edfa1)

**Comment by billypurdue** [comment]
I have the Mesh Agent running on a Windows 10 32 bit host and am experiencing the same behavior.  It seems to crash often while I'm utilizing the agent to remotely access the computer.  I did enabled the generation of the dump file but like the others in this thread there is no dump file being generated.  I saw another thread about Defender being a potential cause of this, however I disabled Defender and installed a third party antivirus solution and it's continuing to happen, even after adding an exception in my third party AV.

**Comment by marek26340** [comment]
Hi, it looks like I also discovered this issue on my 32bit PCs on the network. They run some older apps that still can't run on 64bit OSes reliably. Apologies for having the event viewer log in Czech, feel free to translate it. No core dump is being generated.
I can reproduce this issue pretty reliably if I connect to the PC to the login screen and then log in to a user account. Then agent seems to crash atleast 2 to 3 times afterwards, then it sort-of stabilizes. 
It also seems like the agent can't detect Intel AMT on these systems at all. Been pulling my hair out over AMT not getting detected this whole evening. 

PC: HP Compaq 6300 SFF
Running Windows 10 Pro 19045 32bit on the latest december patches
Intel Core i3 3220, BIOS latest v3.08, ME FW 8.1.72.3002, MEI 8.1.0.1263, LMS 8.1.0.1281, configured manually using MEBx

Název protokolu:Application
Zdroj:         Application Error
Datum:         28.12.2024 18:07:44
ID události:   1000
Kategorie úlohy:(100)
Úroveň:        Chyba
Klíčová slova: Klasické nastavení
Uživatel:      Není k dispozici
Počítač:       ---REDACTED---.cz
Popis:
Název chybující aplikace: MeshAgent.exe, verze: 0.0.0.0, časové razítko: 0x639396df
Název chybujícího modulu: unknown, verze: 0.0.0.0, časové razítko: 0x00000000
Kód výjimky: 0xc0000005
Posun chyby: 0x038ef880
ID chybujícího procesu: 0x1a1c
Čas spuštění chybující aplikace: 0x01db594acbf32f95
Cesta k chybující aplikaci: C:\Program Files\Mesh Agent\MeshAgent.exe
Cesta k chybujícímu modulu: unknown
ID zprávy: d5e3bafe-cceb-4fe6-8805-1e948fbc49fb
Úplný název chybujícího balíčku: 
ID aplikace související s chybujícím balíčkem: 
Kód XML události:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>100</Task>
    <Opcode>0</Opcode>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2024-12-28T17:07:44.3701212Z" />
    <EventRecordID>112728</EventRecordID>
    <Correlation />
    <Execution ProcessID="0" ThreadID="0" />
    <Channel>Application</Channel>
    <Computer>---REDACTED---.cz</Computer>
    <Security />
  </System>
  <EventData>
    <Data>MeshAgent.exe</Data>
    <Data>0.0.0.0</Data>
    <Data>639396df</Data>
    <Data>unknown</Data>
    <Data>0.0.0.0</Data>
    <Data>00000000</Data>
    <Data>c0000005</Data>
    <Data>038ef880</Data>
    <Data>1a1c</Data>
    <Data>01db594acbf32f95</Data>
    <Data>C:\Program Files\Mesh Agent\MeshAgent.exe</Data>
    <Data>unknown</Data>
    <Data>d5e3bafe-cceb-4fe6-8805-1e948fbc49fb</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>

**Comment by KonVei82** [comment]
I am not sure the issue is 32bit specific since i am getting it all the time on all my 64bit systems as well.

<img width="951" height="723" alt="Image" src="https://github.com/user-attachments/assets/22ea4930-16eb-4d57-b3aa-23263fdec6a5" />

**Comment by PrplHaz4** [comment]
> I am not sure the issue is 32bit specific since i am getting it all the time on all my 64bit systems as well.
> <img alt="Image" width="951" height="723" src="https://private-user-images.githubusercontent.com/140968638/655096408-22ea4930-16eb-4d57-b3aa-23263fdec6a5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODk4MzgwMDYsIm5iZiI6MTc4OTgzNzcwNiwicGF0aCI6Ii8xNDA5Njg2MzgvNjU1MDk2NDA4LTIyZWE0OTMwLTE2ZWItNGQ1Ny1iM2FhLTIzMjYzZmRlYzZhNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwOTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDkxOVQxNzA4MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMGViN2VjYWQ3ZjUzNDE1ZDhhNGUyODRmZDNjZTk5ODJiMWY1MzI3MDQ4MWMyNTc3MTljYjE1OWVjYzdkNTdjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.EfJ755zAZAm_twQ23MB9yHtLJHOsDw4aJPTRHsyz4aI">

I don't believe it is either - my screenshot above is from a 64bit system, and it is still happening.

**Comment by si458** [comment]
We have new meshagent builds for you to test which should fix the issue hopefully!

1. Download the artifacts https://github.com/Ylianst/MeshAgent/actions/runs/35435713637
2. Backup old agents from /node_modules/meshcentral/agents/
3. Extract files from the downloaded artifacts and copy them to /node_modules/meshcentral/agents/ overwriting the old ones
4. Add `noAgentUpdate=1` to the `settings` in your config.json
5. Restart meshcentral and wait a min or 2
6. pick a victim pc (sorry a test pc haha)
7. Go into its console tab and click `Agent Actions` then pick `Force Agent Update`
8. Wait for the agent to restart and check it the new version (should show a September 2026 date)
9. Report back any bugs or issues etc!

**Comment by KonVei82** [comment]
So far everything is looking good and no more event ID 1000 on 2 victim pcs.
